src/main/java/janggi/domain/board/Point.java

```diff
@@ -0,0 +1,36 @@
+package janggi.domain.board;
+
+public record Point(int x, int y) {
+    private static final int MINIMUM_ROW = 1;
+    private static final int MAXIMUM_ROW = 10;
+    private static final int MINIMUM_COLUMN = 1;
+    private static final int MAXIMUM_COLUMN = 9;
+
+    public Point {
+        if (isOutOfBoundary(x, y)) {
+            throw new IllegalArgumentException("범위를 벗어났습니다.");
+        }
+    }
+
+    public Point move(Direction direction) {
+        return new Point(x + direction.getX(), y + direction.getY());
+    }
+
+    public boolean isSamePosition(Point point) {
+        return this.x == point.x && y == point.y;
+    }
+
+    public boolean canMove(Direction direction) {
+        int nextX = x + direction.getX();
+        int nextY = y + direction.getY();
+        return nextX >= MINIMUM_ROW && nextX <= MAXIMUM_ROW && nextY >= MINIMUM_COLUMN && nextY <= MAXIMUM_COLUMN;
+    }
+
+    public boolean isOutOfBoundary() {
+        return x > MAXIMUM_ROW || x < MINIMUM_ROW || y > MAXIMUM_COLUMN || y < MINIMUM_COLUMN;
+    }
+
+    private boolean isOutOfBoundary(int x, int y) {
+        return x > MAXIMUM_ROW || x < MINIMUM_ROW || y > MAXIMUM_COLUMN || y < MINIMUM_COLUMN;
+    }
```

Comment:
이렇게 로직이 중복되는 경우에는 다음과 같이 작성할 수도 있어요.
검증 로직이 변경되는 경우에 유지보수하기 쉬워집니다. 
```suggestion
    public boolean isOutOfBoundary() {
        return isOutOfBoundary(this.x, this.y);
    }

    private boolean isOutOfBoundary(int x, int y) {
        return x > MAXIMUM_ROW || x < MINIMUM_ROW || y > MAXIMUM_COLUMN || y < MINIMUM_COLUMN;
    }
```

Reply:
확실히 그렇네요! 중복 제거하겠습니다!